crates/aptos/src/account/key_rotation.rs

```diff
@@ -222,17 +222,48 @@ impl CliCommand<RotateSummary> for RotateKey {
             return Err(CliError::AbortedError);
         }
 
-        let mut profile_config = ProfileConfig {
-            private_key: Some(new_private_key.clone()),
-            public_key: Some(new_private_key.public_key()),
-            account: Some(sender_address),
-            ..self.txn_options.profile_options.profile()?
+        let profile_config = match self.txn_options.profile_options.profile() {
+            Ok(profile) => ProfileConfig {
+                private_key: Some(new_private_key.clone()),
+                public_key: Some(new_private_key.public_key()),
+                account: Some(sender_address),
+                ..profile
+            },
+            Err(_) => {
+                let (rest_url, faucet_url) = if let Some(url) = self.txn_options.rest_options.url {
+                    let url_str = url.to_string();
+                    if url_str == "https://fullnode.testnet.aptoslabs.com" {
+                        (
+                            Some(url_str),
+                            Some("https://faucet.testnet.aptoslabs.com".to_string()),
+                        )
+                    } else if url_str == "https://fullnode.devnet.aptoslabs.com" {
+                        (
+                            Some(url_str),
+                            Some("https://faucet.devnet.aptoslabs.com".to_string()),
+                        )
+                    } else if url_str == "https://fullnode.mainnet.aptoslabs.com" {
+                        (Some(url.to_string()), None)
+                    } else {
+                        // Default to localhost
+                        (Some(url_str), Some("http://localhost:8081".to_string()))
+                    }
+                } else {
+                    return Err(CliError::CommandArgumentError(
+                        "Either [--url or --profile] argument is required".to_string(),
+                    ));
```

Comment:
Don't we essentially already have this logic in the init command? Can you pull that out into a common library? Some kind of Network enum might be the play.